aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -339,7 +339,11 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
               args.lda,
               args.matb->data_ptr<scalar_t>(),
               args.ldb,
+#if defined(USE_ROCM)
+              (&result != &self) ? self.const_data_ptr<scalar_t>() : nullptr,
```

Comment:
Is there a way of writing this more lines of code, but more readable? This isn't exactly clear on a first read and this code is important enough that I worry that sort of complexity will make it confusing.

Reply:
Done.